packages/x-codemod/src/util/removeProps.ts

```diff
@@ -16,6 +16,13 @@ export default function removeProps({ root, componentNames, props, j }: RemovePr
     .find(j.JSXAttribute)
     .filter((attribute) => props.includes(attribute.node.name.name as string))
     .forEach((attribute) => {
-      j(attribute).remove();
+      // Test if we are not remove  a prop of a nested component
+      const attributeParent = attribute.parentPath.parentPath;
+      if (
+        attributeParent.value.type === 'JSXOpeningElement' &&
+        componentNames.includes(attributeParent.value.name.name)
```

Comment:
Shouldn't this be negated, based on the comment above? 🤔

Reply:
Resolved based on https://github.com/mui/mui-x/pull/7698#discussion_r1086459337

Reply:
No, it's the comment that is miss leading. I fixed it